File: pset2/readability/readability.c
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <cs50.h>

int main(void)
{
    // get user input
    string input = get_string("Text: \n");
    float letter = 0;
    float word = 0;
    float sentence = 0;

    for (int i = 0, n = strlen(input); i < n; i++)
    {
        // letters ASCII 65-90 97-122
        if (input[i] > 64 && input[i] < 123)
        {
            letter++;
        }
        // words space=32, !=33 .=46 ?=63
        else if (input[i] == 32 && (input[i - 1] != 33 && input[i - 1] != 46 && input[i - 1] != 63))
        {
            word++;
        }
        else if (input[i] == 33 || input[i] == 46 || input[i] == 63)
        {
            word++;
            sentence++;
        }
    }

    // apply formula index = 0.0588 * L - 0.296 * S - 15.8
    float L = letter * 100 / word;
    float S = sentence * 100 / word;
    float index = round(0.0588 * L - 0.296 * S - 15.8);

    // output
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 15)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int)index);
    }
}

File: pset6/readability/readability.py
from cs50 import get_string
# user input
input = get_string("Text: ")
letter = 0
word = 0
sentence = 0

for i in range(len(input)):
    # letters ASCII 65-90 97-122
    if (ord(input[i]) > 64 and ord(input[i]) < 123):
        letter += 1
    # words space=32, !=33 .=46 ?=63
    elif (ord(input[i]) == 32 and (ord(input[i - 1]) != 33 and ord(input[i - 1]) != 46 and ord(input[i - 1]) != 63)):
        word += 1
    # .!?
    elif (ord(input[i]) == 33 or ord(input[i]) == 46 or ord(input[i]) == 63):
        word += 1
        sentence += 1
# output calculation
L = letter * 100 / word
S = sentence * 100 / word
index = round(0.0588 * L - 0.296 * S - 15.8)

# output
if (index < 1):
    print("Before Grade 1")
elif (index > 15):
    print("Grade 16+")
else:
    print(f"Grade {index}")
